modules/ingester/instance_search.go

```diff
@@ -163,15 +160,36 @@ func (i *instance) searchWAL(ctx context.Context, req *tempopb.SearchRequest, p
 	}
 
 	searchWalBlock := func(b common.WALBlock) {
+		blockID := b.BlockMeta().BlockID.String()
 		span, ctx := opentracing.StartSpanFromContext(ctx, "instance.searchWALBlock", opentracing.Tags{
-			"blockID": b.BlockMeta().BlockID,
+			"blockID": blockID,
 		})
 		defer span.Finish()
 		defer sr.FinishWorker()
 
-		resp, err := b.Search(ctx, req, common.DefaultSearchOptions())
+		var resp *tempopb.SearchResponse
+		var err error
+
+		isTraceQL := false
+		opts := common.DefaultSearchOptions()
+		if api.IsTraceQLQuery(req) {
+			isTraceQL = true
+			// note: we are creating new engine for each wal block,
+			// and engine.Execute is parsing the query for each block
+			resp, err = traceql.NewEngine().Execute(ctx, req, traceql.NewSpansetFetcherWrapper(func(ctx context.Context, req traceql.FetchSpansRequest) (traceql.FetchSpansResponse, error) {
+				return b.Fetch(ctx, req, opts)
+			}))
+		} else {
+			resp, err = b.Search(ctx, req, opts)
+		}
+
 		if err != nil {
-			level.Error(log.Logger).Log("msg", "error searching wal block", "blockID", b.BlockMeta().BlockID.String(), "err", err)
+			msg := "error searching local block"
+			if isTraceQL && errors.Is(err, common.ErrUnsupported) {
```

Comment:
do we need special processing here for traceql and unsupported? i'm fine with just logging the error regardless

Reply:
not really, I did this to make this error more verbose. will just log the error and the tarceql flag.